prow/statusreconciler/controller_test.go

```diff
@@ -867,7 +867,7 @@ func TestControllerReconcile(t *testing.T) {
 		expectErr bool
 	}{
 		{
-			name: "ignored org skips creation, still does retire and migrate",
```

Comment:
This testcase makes it sound like the current behavior was intended. @stevekuznetsov do you recall why the denylist only applies to job creations and not to retirements and migraitons?

Reply:
IIRC the only real use for the deny-list was to not cause thundering herds of new jobs on `kubernetes/kubernetes` - migrating and retiring just use GH tokens.

Reply:
The initial purpose of not causing spamming of new jobs totally makes sense, not quite sure what `migrating and retiring just use GH tokens` means though.

Reply:
There was always enough tokens in order to retire old jobs and migrate contexts - even if the changing jobs were for `kubernetes/kubernetes`. So, these code paths were not part of the deny-list. This is in contrast to compute resources, which is what made triggering new jobs for `kubernetes/kubernetes` impossible.

Reply:
Gotcha. The question is, does `kubernetes/kubernetes` still need the retire and migrate behaviors? If that's the case, then we may need to have more granular control such as breaking `denylist` to the following:

- `denylist-retire`
- `denylist-migrate`
- `denylist-trigger`

Alternatively, we can make `denylist` take in a map, which would be a breaking change though.

Reply:
@stevekuznetsov , any thoughts?

Reply:
I imagine it will be very sorely missed if someone changes a presubmit on `k/k` and the stale contexts are not cleaned up. cc @BenTheElder 

I'd maybe be in favor of a new deny-list field that does everything, and keep the current one as-is?

Reply:
sgtm, added a new option and kept the current one as is. @stevekuznetsov please take another look